autocrit.py

```diff
@@ -0,0 +1,158 @@
+import os
+import random
+import time
+
+import numpy as np
+import openai
+import torch
+from accelerate import Accelerator
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+from tqdm import tqdm, trange
+from transformers import AutoModelForCausalLM, AutoTokenizer, DataCollatorWithPadding
+from itertools import cycle, islice
+
+from rich.console import Console
+
+console = Console(width=80)
+print = console.print
+print0 = lambda *args, **kwargs: print(*args, **kwargs) if os.environ.get("RANK", "0") == "0" else None
+
+@torch.inference_mode()
+def generate(model, tokenizer, prompts, temperature=1, max_new_tokens=256, max_length=2048, stop=[]):
```

Comment:
This should be refactored with the inference backend system (once that is stable)